src/codeql.test.ts

```diff
@@ -942,9 +942,11 @@ test("runTool summarizes several fatal errors", async (t) => {
     async () =>
       await codeqlObject.finalizeDatabase("db", "--threads=2", "--ram=2048"),
     {
-      message:
-        'Encountered a fatal error while running "codeql-for-testing database finalize --finalize-dataset --threads=2 --ram=2048 db". ' +
-        `Exit code was 32 and error was: ${datasetImportError}. Context: ${heapError}. See the logs for more details.`,
+      instanceOf: util.ConfigurationError,
+      message: new RegExp(
+        'Encountered a fatal error while running \\"codeql-for-testing database finalize --finalize-dataset --threads=2 --ram=2048 db\\"\\. ' +
+          `Exit code was 32 and error was: ${datasetImportError}\\. Context: ${heapError}. See the logs for more details\\.`,
```

Comment:
Technically we should also replace the `.`s with `\.`s in the template strings too, though this isn't too concerning for the purposes of a unit test.

```suggestion
        'Encountered a fatal error while running \\"codeql-for-testing database finalize --finalize-dataset --threads=2 --ram=2048 db\\"\\. ' +
          `Exit code was 32 and error was: ${datasetImportError.replaceAll(".", "\\.")}\\. Context: ${heapError.replaceAll(".", "\\.")}\\. See the logs for more details\\.`,
```